scripts/o3de/o3de/manifest.py

```diff
@@ -340,28 +340,37 @@ def get_project_templates(project_path: pathlib.Path) -> list:
 
 # gem.json queries
 def get_gem_gems(gem_path: pathlib.Path) -> list:
-    return get_gems_from_external_subdirectories(get_gem_external_subdirectories(gem_path, set()))
+    return get_gems_from_external_subdirectories(get_gem_external_subdirectories(gem_path, list()))
 
 
-def get_gem_external_subdirectories(gem_path: pathlib.Path, visited_gem_paths: set) -> list:
+def get_gem_external_subdirectories(gem_path: pathlib.Path, visited_gem_paths: list) -> list:
+    '''
+    recursively visit each gems "external_subdirectories" entries and return them in a list
+    :param: gem_path path to the gem whose gem.json will be queried for the "external_subdirectories" field
+    :param: visited_gem_paths stores the list of gem paths visited so far up until this get_path
+    The visited_gem_paths is a list instead of a set to maintain insertion order
+    '''
     if gem_path in visited_gem_paths:
         logger.error(f'A cycle has been detected when visiting external subdirectories at gem path "{gem_path}". The visited paths are: {visited_gem_paths}')
```

Comment:
nit: this could probably be a warning since it doesn't cause any failures

Reply:
The irony of it, was that it doesn't cause any failures because we detect it.
Before we detected the cycle it would cause an infinite loop, which could be classified as an error

Reply:
I have updated the PR with the test from your closed PR.